src/primitives/transaction.h

```diff
@@ -144,24 +146,12 @@ class CTxOut
     bool IsNull() const { return (nValue == -1); }
     uint256 GetHash() const;
 
-    CAmount GetDustThreshold(const CFeeRate &minRelayTxFee) const
+    CAmount GetDustThreshold() const
     {
-        // "Dust" is defined in terms of CTransaction::minRelayTxFee,
-        // which has units satoshis-per-kilobyte.
-        // If you'd pay more than 1/3 in fees
-        // to spend something, then we consider it dust.
-        // A typical spendable txout is 34 bytes big, and will
-        // need a CTxIn of at least 148 bytes to spend:
-        // so dust is a spendable txout less than
-        // 546*minRelayTxFee/1000 (in satoshis)
-        if (scriptPubKey.IsUnspendable())
-            return 0;
-
-        size_t nSize = GetSerializeSize(*this, SER_DISK, 0) + 148u;
-        return 3 * minRelayTxFee.GetFee(nSize);
+        return (CAmount)nDustThreshold;
     }
 
-    bool IsDust(const CFeeRate &minRelayTxFee) const { return (nValue < GetDustThreshold(minRelayTxFee)); }
```

Comment:
didn't we just change the semantics of this functions? I mean before the threshold was dependent on tx ser size now is always 546 satoshi.

Reply:
No not really, it always was 546 satoshi.  After all those calculations and getting of the serialize size, the result was the same  1000/1000 * 546.